runhouse/main.py

```diff
@@ -115,31 +115,52 @@ def notebook(
 @app.command()
 def ssh(cluster_name: str, up: bool = typer.Option(False, help="Start the cluster")):
     """SSH into a cluster created elsewhere (so `ssh cluster` doesn't work out of the box) or not yet up."""
-
     try:
         c = cluster(name=cluster_name)
-    except ValueError:
-        logger.error(
-            f"Could not load cluster called {cluster_name} from Den. Please save it to Den, and rerun."
-        )
-        raise typer.Exit(1)
 
-    if not c.is_shared:
-        if up:
-            try:
-                c.up_if_not()
-            except NotImplementedError:
+        if not c.is_shared:
+            if up:
+                try:
+                    c.up_if_not()
```

Comment:
```suggestion
                    c = c.up_if_not()
```
might be an edge case, but if the cluster isn't actually up and we do bring it back, we don't want to be storing stale IPs, which returning the updated cluster object prevents

Reply:
it looks like `up_if_not` and `up` does directly update the fields of the original cluster object, even if we don't reassign it with the new returned object